### PR TITLE
feat(zero-state): adds zero-state, icon-list and marketing-banner

### PR DESCRIPTION
**Pull request by mattnolting**
As per discussion w/@ryelo, this PR will need some effort from the dashboard maintainers to complete. 

## Prerequisites

- [ ] Scope your styles to your individual card
- [ ] You are using translations when necessary with proper plural/singular modifications
- [ ] You are using the template card provided
- [ ] Use functional components & hooks (please)
- [ ] Attach screenshots (before and after)
- [ ] Make sure that all text in blue is populated with the correct link. If you're unsure what the url should be, just ask!

## What

This PR adds `ZeroState` layout, `IconList` and `MarketingBanner`.

## Mockups
https://marvelapp.com/prototype/8gb8313/screen/70408727

### Before
Screen Shot 2020-11-09 at 10 20 15 AM

### After
Screen Shot 2020-11-09 at 4 36 50 PM
Screen Shot 2020-11-09 at 4 37 00 PM

## Code Review
@christiemolloy @ryelo @AllenBW

## Design Review
@kybaker

### CODE REVIEW

**Review by AllenBW — CHANGES_REQUESTED** [review]
flippin **BEAUTIFUL** @mattnolting!  Ah few general things off the bat...
* need to import dist versions of app deps (especially all the '@patternfly/react-core' ones)
* all strings need to be in Messages.js and then interpolated with intl
* an number of class components should be simplified to being functional components, (will also do away with the `this`)
* it does not appear that your zero state (when the workloadIs=SAP) supports any unique sap messaging, thinking the original message state should be kept until either, official sign off on doing away with sap specific messaging is approved, new sap messaging is approved 🤔 

🤗 🙇

**Comment by AllenBW** [comment]
Ok doing feedback, push up a commit, gotta take a breather, s a lot, followup finishing translations coming up next!

**Comment by AllenBW** [comment]
ok new ss after all mai playing around, as per @kybaker ah few text changes, but mostly the same

#### what it looks like
Screen Shot 2020-11-12 at 8 40 27 AM
Screen Shot 2020-11-12 at 8 40 44 AM
Screen Shot 2020-11-12 at 8 41 12 AM
Screen Shot 2020-11-12 at 8 41 29 AM

**Review by AllenBW — APPROVED** [review]
Yes. 🆗 👌 🙆

File: locales/translations.json [hunk 1]
```diff
@@ -10,10 +10,13 @@
   "complianceTitle": "Compliance",
   "compliantHostCount": "{count, plural, one {# system} other {# systems}}",
   "compliantScore": "{score}% compliant",
+  "contactsales": "Contact sales",
   "critical": "Critical",
   "cvesImpactingSystems": "{cves, plural, one {CVE} other {CVEs}} impacting your systems",
   "dashboardFiltration": "Dashboard filtration temporarily unavailable",
   "dashboardTitle": "Dashboard",
+  "dataPrivacyAndControls": "Red Hat Insights provides a mechanism for users to obtainactionable intelligence regarding suggested improvements todeployed Red Hat software. This document covers thesecurity measures Red Hat puts in place to providesecure transmission, processing, and analysis of thisdata by those tools.",
```
Comment (ryelo) [inline locales/translations.json:18]:
```suggestion
  "dataPrivacyAndControls": "Red Hat Insights provides a mechanism for users to obtain actionable intelligence regarding suggested improvements to deployed Red Hat software. This document covers the security measures Red Hat puts in place to provide secure transmission, processing, and analysis of this data by those tools.",
```

File: locales/translations.json [hunk 2]
```diff
@@ -42,12 +49,18 @@
   "registerYourSystems": "Register your systems",
   "remediationsNoActivity": "Failure",
   "remediationsNoDataBody": "Create an Ansible Playbook to remediate or mitigate vulnerabilities or configuration issues.",
-  "remediationsNoDataTitle": "You haven't created any remediation playbooks yet",
+  "remediationsNoDataTitle": "You haven't created any remediation Playbooks yet",
```
Comment (ryelo) [inline locales/translations.json:52]:
```suggestion
  "remediationsNoDataTitle": "You haven't created any remediation playbooks yet",
```

File: locales/translations.json [hunk 3]
```diff
@@ -42,12 +49,18 @@
...
   "remediationsPlaybookStatus": "{status}",
   "remediationsTotal": "{total} more remediations",
+  "rhm1": "Red Hat Management gives you the flexibility to manage Red Hat Enterprise Linux on-premise or in a hosted environment",
+  "rhm2": "Red Hat Management covers your entire system and security management life cycle",
+  "rhm3": "Red Hat Management includes Red Hat Satellite and the new cloud management services for Red Hat Enterprise Linux",
   "security": "Security",
   "securityAdvisories": "{count, plural, one {security advisory} other {security advisories}}",
+  "securityRedirect": "Understanding our security measures",
+  "singleConsistent": "Single, consistent management solution across on-premise, hybrid cloud, and public cloud. Included with Red Hat Enterprise Linux subscription.",
+  "smartManagement": "Smart management subscription enables push-button remediation of issuesidentified by Insights. Any issues identified by Insights are accompanied byremediation instructions, and with Smart Management, remediation is availableat scale.",
```
Comment (ryelo) [inline locales/translations.json:63]:
```suggestion
  "smartManagement": "Smart management subscription enables push-button remediation of issues identified by Insights. Any issues identified by Insights are accompanied by remediation instructions, and with Smart Management, remediation is available at scale.",
```

File: src/Messages.js [hunk 4]
```diff
@@ -374,5 +374,171 @@ export default defineMessages({
...
+    },
+    contactsales: {
+        id: 'contactsales',
+        description: 'Contact sales',
+        defaultMessage: 'Contact sales'
+    },
+    smartManagement: {
+        id: 'smartManagement',
+        description: 'Smart management subscription',
+        defaultMessage: 'Smart management subscription enables push-button remediation of issuesidentified by Insights. Any issues identified by Insights are accompanied byremediation instructions, and with Smart Management, remediation is availableat scale.'
```
Comment (ryelo) [inline src/Messages.js:406]:
```suggestion
        defaultMessage: 'Smart management subscription enables push-button remediation of issues identified by Insights. Any issues identified by Insights are accompanied by remediation instructions, and with Smart Management, remediation is available at scale.'
```

File: src/Messages.js [hunk 5]
```diff
@@ -374,5 +374,171 @@ export default defineMessages({
...
+    },
+    securityRedirect: {
+        id: 'securityRedirect',
+        description: 'Understanding our security measures',
+        defaultMessage: 'Understanding our security measures'
+    },
+    dataPrivacyAndControls: {
+        id: 'dataPrivacyAndControls',
+        description: 'Data privacy body',
+        defaultMessage: 'Red Hat Insights provides a mechanism for users to obtainactionable intelligence regarding suggested improvements todeployed Red Hat software. This document covers thesecurity measures Red Hat puts in place to providesecure transmission, processing, and analysis of thisdata by those tools.'
```
Comment (ryelo) [inline src/Messages.js:436]:
```suggestion
        defaultMessage: 'Red Hat Insights provides a mechanism for users to obtain actionable intelligence regarding suggested improvements to deployed Red Hat software. This document covers the security measures Red Hat puts in place to provide secure transmission, processing, and analysis of this data by those tools.'
```

File: build/messages/src/Messages.json [hunk 6]
```diff
@@ -59,6 +63,14 @@
     "defaultMessage": "Dashboard",
     "description": "Title of the dashboard"
   },
+  "dataPrivacyAndControls": {
+    "defaultMessage": "Red Hat Insights provides a mechanism for users to obtainactionable intelligence regarding suggested improvements todeployed Red Hat software. This document covers thesecurity measures Red Hat puts in place to providesecure transmission, processing, and analysis of thisdata by those tools.",
```
Comment (ryelo) [inline build/messages/src/Messages.json:67]:
```suggestion
    "defaultMessage": "Red Hat Insights provides a mechanism for users to obtain actionable intelligence regarding suggested improvements to deployed Red Hat software. This document covers the security measures Red Hat puts in place to provide secure transmission, processing, and analysis of this data by those tools.",
```

File: build/messages/src/Messages.json [hunk 7]
```diff
@@ -195,6 +235,18 @@
...
+  "securityRedirect": {
+    "defaultMessage": "Understanding our security measures",
+    "description": "Understanding our security measures"
+  },
+  "singleConsistent": {
+    "defaultMessage": "Single, consistent management solution across on-premise, hybrid cloud, and public cloud. Included with Red Hat Enterprise Linux subscription.",
+    "description": "Single, consistent management solution"
+  },
+  "smartManagement": {
+    "defaultMessage": "Smart management subscription enables push-button remediation of issuesidentified by Insights. Any issues identified by Insights are accompanied byremediation instructions, and with Smart Management, remediation is availableat scale.",
```
Comment (ryelo) [inline build/messages/src/Messages.json:247]:
```suggestion
    "defaultMessage": "Smart management subscription enables push-button remediation of issues identified by Insights. Any issues identified by Insights are accompanied by remediation instructions, and with Smart Management, remediation is available at scale.",
```

File: build/messages/src/Messages.json [hunk 8]
```diff
@@ -103,6 +115,10 @@
     "defaultMessage": "Red Hat Insights for SAP",
     "description": "Red Hat Insights for SAP"
   },
+  "insightsandsmartmanagement": {
+    "defaultMessage": "Insights and smart management",
```
Comment (ryelo) [inline build/messages/src/Messages.json:119]: Should `smart management` be capitalized? It is in other places, so not sure of the casing.

File: locales/translations.json [hunk 9]
```diff
@@ -25,10 +28,14 @@
   "inAdvisor": "in Advisor",
   "incidents": "{incidents, plural, one {# Incident} other {# Incidents}}",
   "insightsForSap": "Red Hat Insights for SAP",
+  "insightsandsmartmanagement": "Insights and smart management",
```
Comment (ryelo) [inline locales/translations.json:31]: Same thing with casing here

File: locales/translations.json (hunk 2 above)
Comment (ryelo) [reply locales/translations.json:52]: playbooks is only capitalized when Ansible precedes it

**Review by ryelo — CHANGES_REQUESTED** [review]

File: build/messages/src/Messages.json (hunk 8 above)
Comment (ryelo) [inline build/messages/src/Messages.json:119]: Caps

File: build/messages/src/Messages.json [hunk 10]
```diff
@@ -195,6 +235,18 @@
...
+  "securityRedirect": {
+    "defaultMessage": "Understanding our security measures",
+    "description": "Understanding our security measures"
+  },
+  "singleConsistent": {
+    "defaultMessage": "Single, consistent management solution across on-premise, hybrid cloud, and public cloud. Included with Red Hat Enterprise Linux subscription.",
+    "description": "Single, consistent management solution"
+  },
+  "smartManagement": {
+    "defaultMessage": "Smart management subscription enables push-button remediation of issues identified by Insights. Any issues identified by Insights are accompanied by remediation instructions, and with Smart Management, remediation is available at scale.",
```
Comment (ryelo) [inline build/messages/src/Messages.json:247]: caps

**Review by ryelo — CHANGES_REQUESTED** [review]

**Comment by AllenBW** [comment]
new ask, kill the cve table, here's what it looks like gone (left in the code cuz we'll wanta use that eventually)

Screen Shot 2020-11-12 at 4 01 31 PM

**Review by AllenBW — APPROVED** [review]
Brilliant! 👍 💯 noicely done all <3

**Pull request merged**